core/src/main/kotlin/cache/data/InteractionData.kt

```diff
@@ -41,31 +45,51 @@ data class InteractionData(
 data class ApplicationCommandInteractionData(
     val id: Snowflake,
     val name: String,
-    val options: Optional<List<ApplicationCommandInteractionDataOptionData>> = Optional.Missing()
+    val options: Optional<List<OptionData>> = Optional.Missing()
 ) {
     companion object {
         fun from(data: DiscordApplicationCommandInteractionData): ApplicationCommandInteractionData {
             return with(data) {
                 ApplicationCommandInteractionData(
                     id,
                     name,
-                    options.mapList { ApplicationCommandInteractionDataOptionData.from(it) })
+                    options.mapList { OptionData.from(it) })
             }
         }
     }
 }
 
 @Serializable
-data class ApplicationCommandInteractionDataOptionData(
-    val name: String,
-    val value: Optional<String> = Optional.Missing(),
-    val options: Optional<List<ApplicationCommandInteractionDataOptionData>> = Optional.Missing()
+data class OptionData(
+        val name: String,
+        val value: Optional<OptionValue<@Serializable(NotSerializable::class) Any?>> = Optional.Missing(),
+        val values: Optional<List<CommandArgument>> = Optional.Missing(),
+        val subCommand: Optional<List<SubCommand>> = Optional.Missing()
 ) {
     companion object {
-        fun from(data: DiscordApplicationCommandInteractionDataOption): ApplicationCommandInteractionDataOptionData {
-            return with(data) {
-                ApplicationCommandInteractionDataOptionData(name, value, options.mapList { from(it) })
+        fun from(data: Option): OptionData = with(data) {
+            when(data) {
+                is SubCommand -> OptionData(name, values = data.options)
+                is CommandArgument -> OptionData(name, value = Optional(data.value))
+                is CommandGroup -> OptionData(name, subCommand = data.options)
             }
         }
     }
-}
\ No newline at end of file
+}
+
+
+/**
+ * A serializer that's sole purpose is to provide a No-Op serializer for [Any].
+ * Since kotlinx.serialization doesn't allow using [Any] in a generic context.
```

Comment:
Should clarify that this is for scenarios where the serialization type is not know and not relevant to the serialization itself.